src/DateTimeParser.jl

```diff
@@ -0,0 +1,512 @@
+)
+const AMPM = Dict{AbstractString, Int}(
+    "am" => 1, "a" => 1,
+    "pm" => 2, "p" => 2,
+)
+const JUMP = (
+    " ", ".", ",", ";", "-", "/", "'", "at", "on", "and", "ad", "m", "t", "of", "st",
+    "nd", "rd", "th", "the",
+)
+const PERTAIN = ("of",)
+const UTCZONE = ("utc", "gmt", "z",)
+
+function parsedate(datetimestring::AbstractString, fuzzy::Bool=false;
+    default::ZonedDateTime=ZonedDateTime(DateTime(today()),localzone()),
+    timezone_infos::Dict{AbstractString, TimeZone} = Dict{AbstractString, TimeZone}(), # Specify what a timezone is
+    dayfirst::Bool=false, # MM-DD-YY vs DD-MM-YY
+    yearfirst::Bool=false, # MM-DD-YY vs YY-MM-DD
+    language::AbstractString="english", # Language in Dates.VALUETOMONTH and VALUETODAYOFWEEK
+)
+    datetimestring = strip(datetimestring)
+
+    if isempty(datetimestring)
+        return default
+    end
+
+    weekday = Dict{AbstractString, Int}()
+    for key in keys(VALUETODAYOFWEEK[language])
+        weekday[lowercase(VALUETODAYOFWEEK[language][key])] = key
+    end
+    for key in keys(VALUETODAYOFWEEKABBR[language])
+        weekday[lowercase(VALUETODAYOFWEEKABBR[language][key])] = key
+    end
+
+    monthtovalue = Dict{AbstractString, Int}()
+    for key in keys(VALUETOMONTH[language])
+        monthtovalue[lowercase(VALUETOMONTH[language][key])] = key
+    end
+    for key in keys(VALUETOMONTHABBR[language])
+        monthtovalue[lowercase(VALUETOMONTHABBR[language][key])] = key
+    end
+
+    res = Dict()
+
+    # year/month/day list
+    ymd = []
+    # Index of the month string in ymd
+    mstridx = -1
+
+    tokens = _parsedatetokens(datetimestring)
+    len = length(tokens)
+    i = 1
+    while i <= len
+        token = tokens[i]
+        tokenlength = length(token)
+        # Check if it's a number
+        if !isnull(tryparse(Float64, tokens[i]))
+            # Token is a number
+            i += 1
+            if length(ymd) == 3 && tokenlength in (2,4) &&
+                (i>=len || (tokens[i] != ":" && !haskey(HMS, lowercase(tokens[i]))))
+                # 19990101T23[59]
+                res["hour"] = parse(Int, token[1:2])
+                if tokenlength == 4
+                    res["minute"] = parse(Int, token[3:4])
+                end
+            elseif tokenlength == 6 || (tokenlength > 6 && search(token, '.') == 7)
+                # YYMMDD or HHMMSS[.ss]
+                if length(ymd) == 0 && !contains(token, ".")
+                    push!(ymd, convertyear(parse(Int, token[1:2])))
+                    push!(ymd, parse(Int, token[3:4]))
+                    push!(ymd, parse(Int, token[5:end]))
+                else
+                    # 19990101T235959[.59]
+                    res["hour"] = parse(Int, token[1:2])
+                    res["minute"] = parse(Int, token[3:4])
+                    temp = parse(Float64, token[5:end])
+                    res["second"] = floor(Int, temp)
+                    res["millisecond"] = round(Int, (temp % 1) * 1000)
```

Comment:
Seems to me that it would be easier to just have the tokens not include floating point numbers. Integers only maybe?